airflow/providers/amazon/aws/operators/redshift_sql.py

```diff
@@ -74,5 +75,12 @@ def get_hook(self) -> RedshiftSQLHook:
     def execute(self, context: 'Context') -> None:
         """Execute a statement against Amazon Redshift"""
         self.log.info(f"Executing statement: {self.sql}")
+        sql_stmts = sqlparse.split(self.sql)
         hook = self.get_hook()
-        hook.run(self.sql, autocommit=self.autocommit, parameters=self.parameters)
+
+        with hook.get_conn() as con:
+            con.autocommit = self.autocommit
+            with con.cursor() as cursor:
+                for stmt in sql_stmts:
+                    cursor.execute(stmt)
+                    if self.autocommit is False: con.commit()
```

Comment:
I don't think this is the right approach.
The functionality should exist in the hook and the operator should utilize it.
The operator should not use cursor directly as you do it here.

See https://github.com/apache/airflow/pull/15533 adding similar functionality for Snowflake

Reply:
@eladkal I went ahead and moved lines 81-86 to the RedshiftSQLHook as a sub method and called it in the RedshiftSQLOperator. Is this the right approach that you were envisioning?

Reply:
Not quite...
The function `run` comes from:
https://github.com/apache/airflow/blob/900bad1c67654252196bb095a2a150a23ae5fc9a/airflow/hooks/dbapi.py#L163
your suggestion to not use it and set `run_sql` means that now the hook has two ways to execute sql. This is not ideal.

You should make it work with `run`. You can override the `run` by having a new implementation in `RedshiftSQLHook` (Just like we do it for Snowflake in the PR I shared)


To be specific:
The operator calling `hook.run()` should stay exactly as it is. Let me also point that by removing `parameters=self.parameters` in the operator you are forcing breaking change which is something that we avoid unless absolutely necessary (and I don't think this is a case where it's necessary correct me if i'm wrong)
 So the hook needs to handle single query or multiple.